worker/src/index.ts

```diff
@@ -32,16 +33,71 @@ function forwardRequestToOpenAi(
 type HeliconeRequest = {
   dbClient: SupabaseClient;
   request: Request;
-
   auth: string;
   body?: string;
+  prompt?: Prompt;
 } & HeliconeHeaders;
 
 interface HeliconeHeaders {
   requestId: string;
   userId: string | null;
   promptId: string | null;
   properties?: Record<string, string>;
+  isPromptRegexOn: boolean;
+}
+
+async function getPromptId(
+  dbClient: SupabaseClient,
+  prompt: Prompt
+): Promise<Result> {
+  // First, get the prompt id if there's a match in the prompt table
+  const { data, error } = await dbClient
+    .from("prompt")
+    .select("id")
+    .eq("prompt", prompt.prompt)
+    .limit(1);
+  if (error !== null) {
+    return { data: null, error: error.message }; 
+  }
+  if (data !== null && data.length > 0) {
+    return { data: data[0].id, error: null };
+  } else {
+
+    // First, query the database to find the highest prompt name suffix
+    const { data: highestSuffixData, error: highestSuffixError } = await dbClient
+    .from("prompt")
+    .select("name")
+    .order("name", { ascending: false })
+    .like("name", "Prompt (%)")
+    .limit(1)
+    .single();
+
+    // Extract the highest suffix number from the highest prompt name suffix found
+    let highestSuffix = 0;
+    if (highestSuffixData) {
+      const matches = highestSuffixData.name.match(/\((\d+)\)/);
+      if (matches) {
+        highestSuffix = parseInt(matches[1]);
+      }
+    }
+
+    // Increment the highest suffix to get the new suffix for the new prompt name
+    const newSuffix = highestSuffix + 1;
+
+    // Construct the new prompt name with the new suffix
+    const newPromptName = `Prompt (${newSuffix})`;
```

Comment:
This is really cool. Do you think we should want to ever let the user define the names themselves? 

Not blocking: but i wonder if we should add an extra header like "Helicone-prompt-name"